packages/snap/src/utils/veramoUtils.ts

```diff
@@ -52,12 +56,34 @@ export async function veramoListVCs(
  * @param {string} challenge - challenge of the VP
  * @returns {Promise<VerifiablePresentation | null>} - generated VP
  * */
+
+type CreateVPRequestParams = {
+  vcs: [
+    {
+      id: string;
+      metadata?: {
+        store?: AvailableVCStores;
+      };
+    }
+  ];
+
+  proofFormat?: string;
+  proofOptions?: {
+    type?: string;
+    domain?: string;
+    challenge?: string;
+  };
+};
 export async function veramoCreateVP(
   params: ApiParams,
-  vcId: string,
-  challenge?: string,
-  domain?: string
+  createVPParams: CreateVPRequestParams
 ): Promise<VerifiablePresentation | null> {
+  const id = createVPParams.vcs[0].id;
+  const store = createVPParams.vcs[0].metadata?.store;
```

Comment:
This is unused and we need to find a better way to handle the querying on different store &rarr; the current try catch approach is not really scalable (and it looks ugly :wink:)

Reply:
But can be done together with the above comment in the next PR